doc/architecture.md

```diff
@@ -1,109 +1,176 @@
 - [Pipeline](#the-pipeline)
   - [Data Types](#data-types)
   - [Plugins](#the-plugins)
+  - [Flow](#flow)
 
 ## The Configuration Files
 These files define the components and data structure that will be used by a running instance of OnAIR.
 
 ### Initialization (.ini) File
-Provides the details about the components and other files used by OnAIR to perform its data reasoning.
+Provides the details about the components and other files used by OnAIR when performing its data reasoning.
+
+>#### **[FILES]** (Required)
+>Pathing and file name information  
+>|Key|Required?|Expected Type|Description|
+>|-|-|-|-|
+>|TelemetryFilePath|Yes|str|location of the telemetry file on the local system|
+>|TelemetryFile|Yes|str|name of the telemetry file, used by file access [OnAir data sources](#the-onairdatasource)|
+>|MetaFilePath|Yes|str|location of the telemetry definitions file|
+>|MetaFile|Yes|str|name of the [telemetry definitions file](#telemetry-definitions-json-file), a JSON file|
+
+>#### **[DATA_HANDLING]** (Required)
+>The telemetry data parser
+>|Key|Required?|Expected Type|Description|
+>|-|-|-|-|
+>|DataSourceFile|Yes|str|full path and python file name of the [OnAir data source](#the-onairdatasource) to use|
+
+>#### **[PLUGINS]** (Required) 
+>The various [plugins](#the-plugins) that OnAIR will use to construct the [pipeline](#the-pipeline)
+>|Key|Required?|Expected Type|Description|
+>|-|-|-|-|
+>|KnowledgeRepPluginDict|Yes|dict|Knowledge Representation plugins to be used at runtime|
+>|LearnersPluginDict|Yes|dict|Learner plugins to be used at runtime|
+>|PlannersPluginDict|Yes|dict|Planner plugins to be used at runtime|
+>|ComplexPluginDict|Yes|dict|Complex Reasoner plugins to be used at runtime|
+>
+>Each plugin dictionary item consists of the key, name for the plugin at runtime, to value, path to the plugin's base directory. Runtime plugin name is not required to be the same as the name of the plugin, but the directory used for the plugin must contain the concrete plugin class named **\<directory name\>_plugin.py**. Multiple of the same plugin directory may be used by defining unique key names, e.g. {"Reporter1":"plugins/reporter", "Reporter2":"plugins/reporter"} will provide two reporter plugins. A plugin type may be left empty by giving it an empty dict, **{}**.
+
+>#### **[OPTIONS]** (Optional)
+>Provides settings for runtime
+>|Key|Required?|Expected Type|Description|
+>|-|-|-|-|
+>|IO_Enabled| No | bool | runtime text output or not, omission equates to **_false_**
+
+### Telemetry (.json) File
+WIP - Section incomplete
+
+
+Defined in the [initilization file](#initialization-ini-file) file as the 'Metafile.' Provides the information necessary for OnAIR to understand received data.
 
-It consists of the following sections:
+## The OnAirDataSource
+This is the adapter that attaches OnAIR to a telemetry source. It provides the basic data that is used by the cognitive components to render reasoning about the system.
```

Comment:
I've started to use the language of "parsers" <- a static data source (i.e. csv_parser.py) and "adapters" <- connects to a live data source (i.e. redis_adapter.py). Both parsers and adapters are data sources.

Reply:
Does that mean you want this sentence changed somehow? Mention both?